builtin/providers/aws/resource_aws_iam_role.go

```diff
@@ -145,22 +146,31 @@ func resourceAwsIamRoleDelete(d *schema.ResourceData, meta interface{}) error {
 	// Loop and remove this Role from any Profiles
 	if len(resp.InstanceProfiles) > 0 {
 		for _, i := range resp.InstanceProfiles {
-			_, err := iamconn.RemoveRoleFromInstanceProfile(&iam.RemoveRoleFromInstanceProfileInput{
+			time.Sleep(500 * time.Millisecond)
```

Comment:
Because the IAM Instance Profile was being deleted between my GetInstanceProfile and the remove itself. Without the sleep, it runs okay a lot of time, but sometimes it fails.